app/services/generate_counter_metrics.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Import counter views and downloads for a given tenant
+# Currently, this only imports the first resource type. Additional work would need to be done for works with multiple resource types.
+# The resource_type field in the counter_metrics table is single value and not an array.
+class GenerateCounterMetrics
+  def self.generate_counter_metrics(ids: 'all', limit: 10)
+    if ids == 'all' || ids.blank?
+      test_works = GenericWork.all.to_a.sample(limit)
+
+      test_works.concat(test_works.sample(limit - test_works.length)) while test_works.length < limit
+
+      message = "Creating test data for #{limit} randomly selected GenericWorks"
+    else
+      test_works = ActiveFedora::Base.where(id: ids).to_a
+      message = "Creating test data for works with ids #{ids}"
+    end
+    Rails.logger.info message
+    test_works.each do |work|
+      worktype = work.class
+      work_id = work.id
+      resource_type = work.resource_type.first
+      date = random_date_in_last_3_months
```

Comment:
Let's not assign a random date...let's have a date range (e.g. the last 3 months) that we loop through:

```ruby
(0..120).each do |integer|
  date =  integer.days.ago.to_date
  total_item_investigations = rand(1..10)
  total_item_requests = rand(1..10)
  Hyrax::CounterMetric.create!(
    ...
  )
```